### PR TITLE
Enable hotplug ceph disk cases

### PR DESCRIPTION
**Pull request by chunfuwen**
Moreover,add rbd port and readonly disk snapshot cases

Signed-off-by: chunfuwen <chwen@redhat.com>

### CODE REVIEW

**Comment by chunfuwen** [comment]
@will-Do 
@balamuruhans 

Please help take a look.

File: libvirt/tests/src/virtual_disks/virtual_disks_ceph.py [hunk 1]
```diff
@@ -110,6 +110,10 @@ def create_pool():
         # pool-autostart --disable
         if not sp.set_pool_autostart(pool_name, "--disable"):
             raise exceptions.TestFail("Failed to set pool autostart")
+        # If port is not pre-configurated, port value should not be hardcoded in pool information.
+        if "yes" == params.get("rbd_port", "no"):
+            if 'port' in virsh.pool_dumpxml(pool_name):
+                raise exceptions.TestFail("port attribute should not be in pool information")
```
Comment (balamuruhans) [inline libvirt/tests/src/virtual_disks/virtual_disks_ceph.py]: let us use test.fail() here and elsewhere accordingly

File: libvirt/tests/src/virtual_disks/virtual_disks_ceph.py [hunk 2]
```diff
@@ -110,6 +110,10 @@ def create_pool():
         # pool-autostart --disable
         if not sp.set_pool_autostart(pool_name, "--disable"):
             raise exceptions.TestFail("Failed to set pool autostart")
+        # If port is not pre-configurated, port value should not be hardcoded in pool information.
```
Comment (balamuruhans) [inline libvirt/tests/src/virtual_disks/virtual_disks_ceph.py]: %s/pre-configurated/pre-configured

File: libvirt/tests/src/virtual_disks/virtual_disks_ceph.py (hunk 1 above)
Comment (chunfuwen) [inline libvirt/tests/src/virtual_disks/virtual_disks_ceph.py]: updated

File: libvirt/tests/src/virtual_disks/virtual_disks_ceph.py (hunk 2 above)
Comment (chunfuwen) [inline libvirt/tests/src/virtual_disks/virtual_disks_ceph.py]: updated

**Comment by chunfuwen** [comment]
@balamuruhans 

new patch had been uploaded.

File: libvirt/tests/src/virtual_disks/virtual_disks_ceph.py [hunk 3]
```diff
@@ -557,21 +567,28 @@ def check_in_vm(vm_obj, target, old_parts, read_only=False):
...
+        if test_blockcopy:
             logging.info("Creating %s...", vm_name)
             vmxml_for_test = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
             if vm.is_alive():
                 vm.destroy(gracefully=False)
             vm.undefine()
             if virsh.create(vmxml_for_test.xml, **virsh_dargs).exit_status:
                 vmxml_backup.define()
-                raise exceptions.TestFail("Cann't create the domain")
+                raise test.fail("Cann't create the domain")
```
Comment (balamuruhans) [inline libvirt/tests/src/virtual_disks/virtual_disks_ceph.py]: %s/Cann't/Can't

Comment (chunfuwen) [inline libvirt/tests/src/virtual_disks/virtual_disks_ceph.py]: Done

**Comment by chunfuwen** [comment]
@balamuruhans 
new patch had been uploaded.

**Comment by balamuruhans** [comment]
Thanks @chunfuwen

**Pull request merged**